plugin-server/src/utils/token-bucket.ts

```diff
@@ -21,22 +21,20 @@ export class Storage {
     }
 
     replenish(key: string, now?: number): void {
-        if (typeof now === 'undefined') {
-            now = Date.now()
-        }
-
-        if (this.buckets.has(key) === false) {
-            this.buckets.set(key, [this.bucketCapacity, now])
+        const replenish_timestamp: number = now ?? Date.now()
+        const bucket = this.buckets.get(key)
+        if (bucket === undefined) {
+            this.buckets.set(key, [this.bucketCapacity, replenish_timestamp])
             return
         }
 
-        // We have checked the key exists already, so this cannot be undefined
-        const bucket: Bucket = this.buckets.get(key)!
-
-        // replenishRate is per second, but timestamps are in milliseconds
-        const replenishedTokens = this.replenishRate * ((now - bucket[1]) / 1000) + bucket[0]
-        bucket[0] = Math.min(replenishedTokens, this.bucketCapacity)
-        bucket[1] = now
+        // Replenish the bucket if replenish_timestamp is higher than lastReplenishedTimestamp
```

Comment:
What does `lastReplenishedTimestamp` refer to? Does the comment need an update?

And nit, but mixing in the snake-cased `replenish_timestamp` is weird. :)